### DISCUSSION TITLE
TPM per transaction type

### DISCUSSION THREAD

**Discussion by abort** [TPC-Council/HammerDB#206] [Q&A] [answered]
Hi,

I am interested in the TPM of each and every transaction type (New-order, Payment, Delivery, Order-status, Stock-level) on a non-clustered Oracle database by using the cli on a timed workload. Currently trying to understand the best approach to this as I only acquire TPM/NOPM at the end of the test, which is not per transaction. I am interested in a windowed approach as I am doing some database maintenance during the test and would like to know the implications on the performance of those actions. 

My current approach has been changing the code a bit to execute inserts in a log table (in an autonomous transaction) after each procedure, such that in hindsight I can compute the TPM / transaction on each timestamp. Is there anything already in place for this that I am not aware of or do you have other suggestions how to go about this?

**Comment by sm-shaw** [comment, marked as answer]
Hi,
Yes as a starting point something like this does exist in the Connect Pool functionality.  This is designed for connecting to multiple instances of a cluster at the same time, however I can't see it stopping you from configuring it to run on a cluster of 1 if you wish. 
When you choose connect pool then it triggers a client side transaction count. This is in case your server does not capture NOPM/TPM across all instances. it will report the average transactions per minute but also report exactly how many each virtual user actually did. 
```
Vuser 1:2 Active Virtual Users configured
Vuser 1:TEST RESULT : System achieved 26207 NOPM from 180515 SQL Server TPM
Vuser 1:CLIENT SIDE TPM : neworder 78319 payment 78185 delivery 7855 stocklevel 7826 orderstatus 7809

Vuser 2:VU2 processed neworder 275335 payment 273822 delivery 27495 stocklevel 27588 orderstatus 27568 transactions
Vuser 3:VU3 processed neworder 272901 payment 273475 delivery 27493 stocklevel 27194 orderstatus 27097 transactions
```

**Reply by abort** [reply]
Thanks for that pointer! I just have had a look, it might work with some tweaking by resetting the counters every minute. As i am specifically looking into a solution with a sliding window (e.g. at t = 100s ->  order tpm = X and at t=140 -> order tpm = Y). This allows me to see the effects of the maintenance i am doing on one of the tables in the very same run.
